libraries/chain/include/eosio/chain/block.hpp

```diff
@@ -129,3 +142,20 @@ FC_REFLECT_DERIVED(eosio::chain::transaction_receipt, (eosio::chain::transaction
 FC_REFLECT(eosio::chain::additional_block_signatures_extension, (signatures));
 FC_REFLECT(eosio::chain::quorum_certificate_extension, (qc));
 FC_REFLECT_DERIVED(eosio::chain::signed_block, (eosio::chain::signed_block_header), (transactions)(block_extensions) )
+
+namespace fc::raw {
+   template <typename Stream>
+   void unpack(Stream& s, eosio::chain::signed_block& v) {
+      try {
+         if constexpr (requires { s.extract_mirror(); }) {
```

Comment:
Don't quite get this code. Why the `if constexpr` check? Why the `4096` below?

Reply:
Because if `signed_block` is being unpacked with a `datastream_mirror` then we can use the `extract_mirror()` to populate `signed_block::packed_block`. If a different Stream type is being used, then wrap it in a `datastream_mirror` to capture the packed block data. The 4096 is used to reserve a size in the vector; open to removing the `+ 4096` if we think it might be wasteful.